pkg/exporter/sumologicexporter/sender.go

```diff
@@ -519,59 +510,65 @@ func (s *sender) sendOTLPLogs(ctx context.Context, ld pdata.Logs) error {
 }
 
 // sendNonOTLPMetrics sends metrics in right format basing on the s.config.MetricFormat
-func (s *sender) sendNonOTLPMetrics(ctx context.Context, flds fields) ([]metricPair, error) {
+func (s *sender) sendNonOTLPMetrics(ctx context.Context, rm pdata.ResourceMetrics, flds fields) ([]pdata.Metric, error) {
 	if s.config.MetricFormat == OTLPMetricFormat {
 		return nil, fmt.Errorf("Attempting to send OTLP metrics as non-OTLP data")
 	}
 
 	var (
-		body           bodyBuilder
+		body           bodyBuilder = newBodyBuilder()
 		errs           []error
-		droppedRecords []metricPair
-		currentRecords []metricPair
+		droppedRecords []pdata.Metric
```

Comment:
what about resource attributes?

Reply:
We're sending metrics from a particular resource (see the argument to this func: `pdata.ResourceMetrics`) so no need to set its attributes aside because we'll have it readily available at the call site.

https://github.com/SumoLogic/sumologic-otel-collector/blob/726bdaa6960e7710efd8c4c85a156846407e3cef/pkg/exporter/sumologicexporter/exporter.go#L334-L337